woocommerce/google-listings-and-ads
# Edit Free Listings - Remove corresponding shipping rate and shipping time when target audience is changed

### Changes proposed in this Pull Request:

Closes #1304 .

In this PR, when users remove a target audience country in Edit Free Listings, the corresponding shipping rate and shipping time will also be removed.

### Screenshots:

📹  Demo video with my voice demonstrating changing selected countries:

https://user-images.githubusercontent.com/417342/162034256-0aca8494-5719-47cc-9c38-4b755527e837.mp4

📹  Demo video with my voice demonstrating changing target audience location from 'all' to 'selected':

https://user-images.githubusercontent.com/417342/162438798-9b8d85de-7daa-4493-a7bc-851c1dac9a87.mov

### Detailed test instructions:

Changing individual selected countries:

1. Go to Edit Free Listings. Without making any changes, look at both Step 1 and Step 2 (shipping rates and shipping times), and determine the country that you want to remove.
2. Go to Step 1. Remove the country from the target audience list.
3. Go to Step 2. The country that has been removed from the target audience should not be in the shipping rates and shipping times.
4. Save changes. Reload the page. Make sure the settings are saved and loaded correctly.

Changing target audience location:

1. Go to Edit Free Listings. In Step 1, "selected countries only" should be selected. Take note of the selected countries.
2. Change the option to "All countries".
3. Go to Step 2. You would need to add in more rates and times for more countries. Add some countries in.
4. Go back to Step 1, change the option back to "selected countries only". The selected countries should be same as the countries you took note of earlier.
5. Go to Step 2. The rates and times you added should have been removed, because they are no longer applicable.
6. Save changes. Reload the page. Make sure the settings are saved and loaded correctly.

### Additional details:

With this PR, the countries in shipping rates and shipping times should always be the same as (or a subset of) the target audience countries. We can then simplify some of the related code such as this one in `EditRateModal`:

https://github.com/woocommerce/google-listings-and-ads/blob/0ad357a9e51c315bdea95a50cd38713042eded75/js/src/components/shipping-rate-section/estimated-shipping-rates-card/edit-rate-modal.js#L33-L36


### Changelog entry

> Fix - Remove corresponding shipping rate and shipping time when target audience is changed.
